Added GPS and HB data variables in their classes

ganeshram93 opened the pull request:
1. Added getter functions for GPS and Heartbeat sensor and data
variables inside their respective classes
2. Changed variable name in tcp_server.py

Signed-off-by: ganeshram2493 <ganeshram2493@gmail.com>

ganeshram93 committed:
Added GPS and HB data variables in their classes

1. Added getter functions for GPS and Heartbeat sensor and data
variables inside their respective classes
2. Changed variable name in tcp_server.py

Signed-off-by: ganeshram2493 <ganeshram2493@gmail.com>

ganeshram93 committed:
Deleted invalid commit settings.json

Deleted invalid commit settings.json

coder137 commented on HelloWorld_ST_Sensors/main.cpp:
```diff
@@ -152,16 +143,14 @@ int main() {
 
     // Heartbeat Sensor
-    const heartbeat::bioData hb_data = heartbeat::get_hb_data();
+    const heartbeat::bioData hb_data = hb_obj.get_hb_data();
```
const heartbeat::bioData `&` hb_data = hb_obj.get_hb_data();

ganeshram93 replied:
Updated in new PR.

coder137 commented on HelloWorld_ST_Sensors/user/gps/gps.hpp:
```diff
@@ -23,31 +23,29 @@ enum GPRMC_NMEA_STRING_INDEX_DEFINITION {
   }
 
-  void register_func(void (*external_fn)(void));
+  gps_coordinates_s &get_gps_coordinates(void);
```
`const gps_coordinates_s &`

ganeshram93 replied:
Updated in new PR.

coder137 commented on HelloWorld_ST_Sensors/user/heartbeat/heartbeat.hpp:
```diff
@@ -194,9 +196,8 @@ class sparkfun_MAX32664 {
   const uint8_t SET_FORMAT = 0x00;
   const uint8_t ENABLE = 0x01;
   const uint8_t WRITE_SET_THRESHOLD = 0x01;
+  struct bioData body;
```
no struct keyword

ganeshram93 replied:
Updated in new PR.

coder137 commented on HelloWorld_ST_Sensors/user/internet/internet.cpp:
```diff
@@ -98,24 +101,34 @@ static void update_co_data(spec::CarbonMonoxide *co) {
+}
+
+static void update_gps_data(gps::adafruit_PA6H *gps) {
+  gps::gps_coordinates_s gps_data = gps->get_gps_coordinates();
```
const gps_coordinates_s &

coder137 commented on HelloWorld_ST_Sensors/user/slave/i2c_slave.cpp:
```diff
@@ -273,13 +273,13 @@ void SlaveCommunication::send_heartbeat_data() {
 void SlaveCommunication::send_gps_coordinates_data() {
   uint8_t tx_data[ARR_SIZE * 2] = {0};
-  gps::gps_coordinates_s data = gps::get_gps_coordinates();
+  gps::gps_coordinates_s data = gps_.get_gps_coordinates();
```
const gps_coordinates_s &

ganeshram93 replied:
Updated in new PR.

ganeshram93 committed:
Added const qualifiers and code cleanup

1. Added const qualifiers for HB and GPS variables
2. Cleaned dead code
3. Made default DEBUG_PRINTF to 0 in i2c_slave.cpp
4. Updated some comments

Signed-off-by: ganeshram2493 <ganeshram2493@gmail.com>
